model/src/main/java/org/cloudfoundry/identity/uaa/zone/Links.java

```diff
@@ -98,16 +98,26 @@ public Logout setWhitelist(List<String> whitelist) {
     }
 
     public static class SelfService {
-        private boolean selfServiceLinksEnabled = true;
```

Comment:
selfServiceLinksEnabled is present in the configuration of the existing UAA zones (zone config is stored in the database as text blob).  How are we going to support backward compatibility?

Whenever zone configuration is read from the database, its selfServiceLinksEnabled and links should be parsed into assigning selfServiceResetPasswordEnabled and selfServiceCreateAccountEnabled.  Please review https://devcloud.swcoe.ge.com/devspace/display/RMOCM/Enable+self-service+links+for+UAA+instance to see different configurations.

We haven't discussed backward compatibility in the story.  So you current implementation and the current story could be about new zones and support for existing zones could be handled as another story.  For now selfServiceLinksEnabled can not be deleted.  I am not sure it should be kept here in this class, but somewhere where config is read from the database.

New zones are:
- provisioned by Service Broker "cf cs ..." - selfServiceLinksEnabled flag and links are set to default
- provisioned ad-hoc, for example, during acceptance test setup "POST /identity-zones" - selfServiceLinksEnabled flag and links are set to default unless explicitly specified in the request.

Reply:
@irinaepshteyn , As discussed, we will have another story for support backward compatibility. 
For new zones, split "self-service" flag in UAA zone config has been implemented.

Reply:
Added US554082 - [UAA] Apply split "self-service" flag to existing UAA zone configs